File: gtmworks_ui/src/services/LeadService.js
import http from "../http-common"; 

class LeadService {
  getAllLeads(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/lead/leads`, searchDTO);
  }

  get(leadId) {
    return this.getRequest(`/lead/${leadId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/lead?field=${matchData}`, null);
  }

  addLead(data) {
    return http.post("/lead/addLead", data);
  }

  update(data) {
  	return http.post("/lead/updateLead", data);
  }
  
  uploadImage(data,leadId) {
  	return http.postForm("/lead/uploadImage/"+leadId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new LeadService();

File: src/main/java/com/gtmworks/service/LeadService.java
package com.gtmworks.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.gtmworks.domain.Lead;
import com.gtmworks.dto.LeadDTO;
import com.gtmworks.dto.LeadSearchDTO;
import com.gtmworks.dto.LeadPageDTO;
import com.gtmworks.dto.LeadConvertCriteriaDTO;
import com.gtmworks.service.GenericService;
import com.gtmworks.dto.common.RequestDTO;
import com.gtmworks.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface LeadService extends GenericService<Lead, Integer> {

	List<Lead> findAll();

	ResultDTO addLead(LeadDTO leadDTO, RequestDTO requestDTO);

	ResultDTO updateLead(LeadDTO leadDTO, RequestDTO requestDTO);

    Page<Lead> getAllLeads(Pageable pageable);

    Page<Lead> getAllLeads(Specification<Lead> spec, Pageable pageable);

	ResponseEntity<LeadPageDTO> getLeads(LeadSearchDTO leadSearchDTO);
	
	List<LeadDTO> convertLeadsToLeadDTOs(List<Lead> leads, LeadConvertCriteriaDTO convertCriteria);

	LeadDTO getLeadDTOById(Integer leadId);







}





